webapp/src/ts/services/select2-search.service.ts

```diff
@@ -77,17 +83,39 @@ export class Select2SearchService {
     this.searchService
       .search('contacts', filters, searchOptions)
       .then((documents) => {
-        if (currentQuery === params.data.q) {
```

Comment:
This would always be `true` because `currentQuery` is defined and assigned a value in the same scope here. By pulling it into a class field, it can now be used to discard outdated results